### PR TITLE
Allow for a test mode

### PR DESCRIPTION
**Pull request by agocs**
I refactored out the `cacheDB` global, broke the beaglebone stuff out into its own `Robot` struct with some methods on it, created a `Robotter` interface which `Robot` implements, then I created a `DummyRobot` that also implements `Robotter`. 

Build using `go build main.go beagleboneRobot.go dummyRobot.go server.go`

Run using `./main -testMode=true`

### CODE REVIEW

File: beagleboneRobot.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	return string(buf[1 : n-3])
+}
+
+func (r Robot) runRobot() {
+	defer r.cacheDB.Close()
+	for {
+
+		code := r.serialRead()
+
+		// I'm not 100% sure what this code is doing. Is it initalizing `RFIDBucket` if it doesn't exist?
```
Comment (agocs) [inline beagleboneRobot.go]: @loansindi Could you confirm my guess about this block?

**Comment by agocs** [comment]
cc @loansindi @codersquid

File: beagleboneRobot.go (hunk 1 above)
Comment (loansindi) [inline beagleboneRobot.go]: @tachoknight wrote this, but your guess looks right

**Comment by agocs** [comment]
@loansindi I removed the dependency from members.ps1.org from the code you sent me and put it into the dummy robot.

File: dummyRobot.go [hunk 2]
```diff
@@ -20,7 +21,13 @@ func (d DummyRobot) runRobot(shutdown chan bool) {
 			log.Println("Caught the shutdown signal. Bailing out.")
 			goto quit
 		default:
-			time.Sleep(1 * time.Second)
+
+			reader := bufio.NewReader(os.Stdin)
+			fmt.Print("Code: ")
+			text, _ := reader.ReadString('\n')
+			urlString := "https://members.pumpingstationone.org/rfid/check/FrontDoor/" + text //let's remove a dependency on members.ps1.org
+			fmt.Printf("I would send a request to %v but let's assume it just works.\n", urlString)
```
Comment (loansindi) [inline dummyRobot.go]: We may also need to test failure features. Bonus points if we get to pick the http response code, now that I think about it.

Comment (agocs) [inline dummyRobot.go]: Not right now. We need to break the "is this code correct?" feature out of the real robot so we can test it independently of members.ps1.org and the BoltDB. It doesn't make any sense to put any logic in this fakebot that really only exists as a test harness right now.

Comment (agocs) [inline dummyRobot.go]: I think the scope of this PR has grown large enough, and further refactoring should be a future project.

Comment (loansindi) [inline dummyRobot.go]: Okay, that's a reasonable view.

Comment (codersquid) [inline dummyRobot.go]: Is this ready to merge to master?

Comment (loansindi) [inline dummyRobot.go]: lgtm

**Pull request merged**